### ISSUE TITLE
Parser error when using TypeScript 4.7 Instantiation Expressions

### ISSUE THREAD

**Issue by OliverJAsh** [trivago/prettier-plugin-sort-imports#156]
**Your Environment**
 * **Prettier version**: 2.7.0
 * **node version**: 16.3.0
 * **package manager**: yarn@1.22.17
 * **IDE**: N/A

**Describe the bug**

Parser error when using TypeScript 4.7 Instantiation Expressions ([supported in Prettier 2.7.0](https://prettier.io/blog/2022/06/14/2.7.0.html#instantiation-expressionshttpsdevblogsmicrosoftcomtypescriptannouncing-typescript-4-7instantiation-expressions)).

**To Reproduce**

Using the latest version of `prettier` and `@trivago/prettier-plugin-sort-imports` at the time of writing:

`package.json`:

```
{
  "dependencies": {
    "@trivago/prettier-plugin-sort-imports": "^3.2.0",
    "prettier": "^2.7.0"
  }
}
```

`file.ts`:

```
interface Box<T> {
  value: T;
}
function makeBox<T>(value: T) {
  return { value };
}
const makeHammerBox = makeBox<Hammer>;
const makeWrenchBox = makeBox<Wrench>;
```

```sh-session
$ prettier --parser typescript file.ts
file.ts[error] file.ts: SyntaxError: Unexpected token (7:37)
[error]   1 | interface Box<T> {
[error]   2 |   value: T;
[error]   3 | }
[error]   4 | function makeBox<T>(value: T) {
[error]   5 |   return { value };
[error]   6 | }
[error]   7 | const makeHammerBox = makeBox<Hammer>;
[error]   8 | const makeWrenchBox = makeBox<Wrench>;
```

**Expected behavior**

No error

---

The error only occurs when `@trivago/prettier-plugin-sort-imports` is installed—if remove uninstall it, the parser issue disappears. To make issues like this easier to debug in the future, I would suggest improving the error output so to make it clearer that this error originates from the parser inside of `@trivago/prettier-plugin-sort-imports` and not the parser inside of `prettier`.

**Comment by OliverJAsh** [comment]
I just encountered the same issue trying to use TypeScript 4.9's `satisifies` operator.

**Comment by taiwabisabi** [comment]
Same as @OliverJAsh with TypeScript 4.9's and satisifies operator.

**Comment by OliverJAsh** [comment]
I think the best way to solve this issue would be by solving https://github.com/trivago/prettier-plugin-sort-imports/issues/141.

**Comment by byara** [comment]
The issue should be resolved in v4.1.0.

**Comment by OliverJAsh** [comment]
> I think the best way to solve this issue would be by solving #141.

@byara Do you have any thoughts on https://github.com/trivago/prettier-plugin-sort-imports/issues/141? I fear it's only a matter of time before this issue happens again, so I was hoping we could find a more permanent fix.
